src/pdn/src/PdnGen.tcl

```diff
@@ -3543,11 +3545,14 @@ proc generate_stripes {tag net_name} {
     if {[dict exists $grid_data straps $lay width]} {
       set area [dict get $grid_data area]
       # debug "Area $area"
+      # Calculate the numebr of rings of core_domain
+      set ring_number [expr [llength [get_voltage_domain_power [dict get $design_data core_domain]]] + [llength [get_voltage_domain_ground [dict get $design_data core_domain]]]] 
       if {[dict exists $grid_data core_ring] && [dict exists $grid_data core_ring $lay]} {
-        set area [adjust_area_for_core_rings $lay $area]
+        set area [adjust_area_for_core_rings $lay $area $ring_number]
```

Comment:
The calculation of the number of stripes should be done inside adjust_are_for_core_rings, there is no need to have an extra parameter for the function.

Reply:
The `adjust_area_for_core_rings` now has two parameters which are `layer `and `area`. I am a bit confused how to calculate the number of power rings of a voltage domain with the existing parameters? Could you give me some hints? Thank you.